cmake/linux/package_linux.sh.in

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+# Creates Linux ".AppImage" for @PROJECT_NAME_UCASE@
+
+set -e
+
+# Console colors
+RED="\\x1B[1;31m"
+GREEN="\\x1B[1;32m"
+YELLOW="\\x1B[1;33m"
+PLAIN="\\x1B[0m"
+
+USERBIN="$HOME/bin"
+LINUXDEPLOYQT="$USERBIN/linuxdeployqt"
+APPDIR="@CMAKE_BINARY_DIR@/@PROJECT_NAME_UCASE@.AppDir/"
+
+# TODO: Since DESTDIR is set at build time, the true install location is unknown.
+INSTALL="@CMAKE_BINARY_DIR@/../target"
+
+# linuxdeployqt assumes installations contain /usr/ prefixes.
+# 1. Make a directory containing /usr
+#    > mkdir -p ../target/usr
+# 2. Configure the install prefix for /usr only without "target"
+#    > cmake .. -DCMAKE_INSTALL_PREFIX=/usr -DWANT_QT5=True
+# 3. Use the DESTDIR variables to install to "target" instead
+#    > make DESTDIR=../target/ -j4 install
+make DESTDIR=../target/ -j4 install
+case "@CMAKE_INSTALL_PREFIX@" in 
+  */usr);;
+  *) echo -e "${RED}ERROR: linuxdeployqt requires install prefix of /usr: 
+	Needs:  mkdir ../target/usr	
+	Needs:  CMAKE_INSTALL_PREFIX=/usr
+	Needs:  make DESTDIR=../target
+
+	Wrong:  CMAKE_INSTALL_PREFIX=@CMAKE_INSTALL_PREFIX@	
+${PLAIN}"; exit 1;;
+esac
+
+mkdir -p "$USERBIN"
+
+# Fetch portable linuxdeployqt
+# TODO: Cache and only updated as needed
+echo -e "${GREEN}Downloading linuxdeployqt to ${LINUXDEPLOYQT}${PLAIN}..."
+wget https://github.com/probonopd/linuxdeployqt/releases/download/continuous/linuxdeployqt-continuous-x86_64.AppImage -O "$LINUXDEPLOYQT" -q
+chmod +x "$LINUXDEPLOYQT"
+echo "     [success]"
+
+# Make skeleton AppDir
```

Comment:
So where would you ideally like to install? The reasonis that (at least for some software), the path specified with `DCMAKE_INSTALL_PREFIX` ends up in the compiled binaries, which is what I generally want to avoid, hence recommending `/usr` as the generic prefix. This is not a hard requirement though and you are free to use another prefix.